design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,192 @@
+
+#### From a view navigation item
+
+- In the `Schema Browser` view, a user opens the contextual menu for a view navigation item and selects the option `Open in Data Explorer`. This action will open `Data Explorer` and display the properties of the selected view.
+
+## Base Table
+
+Views require a base table to be set so that the available columns and links can be listed for selection.
+
+### User sets the base table for a new view
+
+- For users starting a new view from scratch, `Base Table` selection will be the first required step.
+- Once `Data Explorer` is launched the user will be prompted to make a selection. The system will list all tables from the active schema. See *Wireframe 1* for an example of the base table selection prompt.
+- Once the `Base Table` is selected, a second step for `Select Columns` is automatically added and the user is prompted to continue with the view creation process.
+- At this point, the result table is empty.
+
+[Wireframe 1](https://share.balsamiq.com/c/jp7WFu663j6sbfKNknuq5Z.png)
+
+### User inspects the base table for an existing view in Data Explorer
+
+- When inspecting an existing view, a user can find the `Base Table` as the first step of the `Workflow Sidebar`.
+- The `Base Table` step cannot be removed, only edited, causing all subsequent steps to be cleared.
+
+[Wireframe 2](https://share.balsamiq.com/c/wTEzQcBamoBBWBSSF5Cmuy.png)
+
+### User opens a view where the base table is missing
+
+- If a user deletes the base table for a view, they should be able to open the view in `Data Explorer` and either resolve the error by replacing the base table with an identical one or clear the workflow and start over.
+
+## Worflow: Column Selection
+
+In the column selection step, users select columns from or linked to the base table of a view.
+
+### User adds a direct column
+
+- Direct columns will be those with values contained directly in the `Base Table` rather than linked from other tables via foreign key relationships.
+- To add a direct column, the user can click on the `Add` button inside a `Select Column` workflow item and click on any column listed under the `Base Table Columns` section.
+- Direct columns will always be listed at the top of the column list in the same order as the base table.
+
+[Wireframe 3](https://share.balsamiq.com/c/hNWzHEM9ksGGp5ZcPixLrA.png)
+
+### User adds a linked column
+
+- To add a linked column, the user can click on any column listed under the `Base Table Links` or `Linked to Base Table` sections. Linked columns are listed under the tables for which a link to or from the `Base Table` exists.
+- The linked column can then be expanded to show all possible values for that field. The default value points to the primary key of the referenced table, however, the user might choose to display values from any other column.
+
+[Wireframe 4](https://share.balsamiq.com/c/hNWzHEM9ksGGp5ZcPixLrA.png)
+
+### User adds multiple select column steps
```

Comment:
I'm still not convinced about allowing multiple steps for column selection. I think it would be better to allow users to add columns in a single step and dynamically update what columns are available to use based on the columns that have already been added.

For example, imagine user adds direct column A. We would only show base table columns + linked columns as options as available columns to use. 

Then, imagine they add formula column B. Then, we show base table columns + linked columns + column A from the query as available columns to use.

Then, imagine they add formula column C. Then, we show base table columns + linked columns + column A and B from the query as available columns to use.

I think this is clearer than having multiple select column steps.

Reply:
@kgodey I understand that we want to simplify, but I feel that we are mixing two different models here... the "step based" and this more "dynamic list" approach.
My concern, is that we need to persist the original selection somewhere, as those are feeding into the next steps. In the same way, a formula column created in step 8 for example, cannot be selected in step 1.

[1. Set base table ]
[2. Select Columns   movie.title.  movie_crew.actorId(as person.full_name)  ]
[3. Add Formula:  actorInitials.  text from:  movie_crew.actorId(as person.full_name)]
[4. Summarize by ActorInitials  and aggregate sum(movie.title)]

I now have a table with Initials and a count of movies per actor name initial. I no longer want the person full name

[5. Select Columns : Actor Initials and sum(movie.title)]

I want to filter the actor name list, before they were initials... at this point I no longer have access to that column, I need to filter before the column selection.

In the dynamic list approach, everything would always be listed, even in steps before they are created (in the case of formulas) so it would stop being a true step based UX. I feel those inconsistencies will confuse users more than multiple select steps. And I'm not sure how, if we don't follow the step based solutions, we'll be able to have more than one filter or summary step.

Reply:
Not responding here since @ghislaineguerin and I discussed the overall flow on a call.